### PR TITLE
Fixed #26308 -- Prevented crash with binary URLs in is_safe_url()

### PR DESCRIPTION
**Pull request by claudep**
This fixes a regression introduced by c5544d28923.

### CODE REVIEW

File: docs/releases/1.9.4.txt [hunk 1]
```diff
@@ -4,9 +4,10 @@ Django 1.9.4 release notes
 
 *Under development*
 
-Django 1.9.4 fixes several bugs in 1.9.3.
+Django 1.9.4 fixes a regression in the 1.9.3 security release.
```
Comment (timgraham) [inline docs/releases/1.9.4.txt]: If you feel this should be released asap, then let's just say "a regression on Python 2 in the 1.9.3 security release where utils.http.is_safe_url() crashes on bytestring URLs (`:ticket:`...)" and remove the "Bugfixes" section.

Comment (claudep) [inline docs/releases/1.9.4.txt]: I find the issue rather critical for any Python 2 deployment, don't you?

Comment (timgraham) [inline docs/releases/1.9.4.txt]: Yes

File: django/utils/http.py [hunk 2]
```diff
@@ -290,6 +290,7 @@ def is_safe_url(url, host=None):
         url = url.strip()
     if not url:
         return False
+    url = force_text(url, errors='replace')
```
Comment (timgraham) [inline django/utils/http.py]: Do you still want to remove this when dropping Python 2? If there isn't something like `six.PY2` will we remember? I guess `force_text()` still has some value in resolving lazy objects but that's not why it's used here. Just wondering...

File: docs/releases/1.8.11.txt [hunk 3]
```diff
@@ -4,9 +4,5 @@ Django 1.8.11 release notes
 
 *Under development*
```
Comment (timgraham) [inline docs/releases/1.8.11.txt]: If you want to add `*March 4, 2016*`, that'll save me a commit.

Comment (claudep) [inline docs/releases/1.8.11.txt]: Now I see that there is a new release blocker for 1.8 (#26324). Would you like to ship it in this release too?

Comment (timgraham) [inline docs/releases/1.8.11.txt]: I think it can wait until the regular monthly bugfix release.

**Comment by timgraham** [comment]
buildbot, test this please.

File: django/utils/http.py [hunk 4]
```diff
@@ -290,6 +290,8 @@ def is_safe_url(url, host=None):
         url = url.strip()
     if not url:
         return False
+    if six.PY2:
+        url = force_text(url, errors='replace')
```
Comment (nedbat) [inline django/utils/http.py]: I'm wondering about `errors='replace'` here.  The function below considers control characters.  Is it possible that some of the bad characters we're trying to detect here will be turned into replacement characters, and then overlooked by `_is_safe_url` below?  The Unicode replacement character is U+FFFD, which is category "So".

Comment (timgraham) [inline django/utils/http.py]: The control character issue is that `<a href="\x08//example.com">` will redirect to example.com but `is_safe_url()` would consider it safe before 011a54315e46acdf288003566b8570440f5ac985. 
In Firefox and Chrome, `<a href="\ufffd//example.com">` will be treated as path relative and redirect to something like `http://localhost:8000/page-where-link-appears/%EF%BF%BD//example.com`, so I don't see a problem.

**Comment by timgraham** [comment]
buildbot, test this please.

**Comment by timgraham** [comment]
For anyone patching other distributions of Django, this fix was amended with 552f03869ea7f3072b3fa19ffb6cb2d957fd8447.

**Pull request merged**